addons/sale/models/sale.py

```diff
@@ -1730,7 +1730,7 @@ def product_id_change(self):
             uom=self.product_uom.id
         )
 
-        vals.update(name=self.get_sale_order_line_multiline_description_sale(product))
+        vals.update(name=self.with_context(lang=self.order_id.partner_id.lang).get_sale_order_line_multiline_description_sale(product))
```

Comment:
Maybe use `get_lang` everywhere to be coherent ? (to be sure the lang of the product context and the lang of the order_line context are the same).

Reply:
```suggestion
lang = get_lang(self.env, self.order_id.partner_id.lang).code
product = self.product_id.with_context(
    lang=lang,
    ...
)
vals.update(name=self.with_context(lang=lang).get_sale_order_line_multiline_description_sale(product))
```